packages/sources/superstate/src/transport/nav.ts

```diff
@@ -0,0 +1,201 @@
+      // If the most recent update from DP is not from previous working day, the data might be stale
+      const expectedDate = getPreviousNonWeekendDay(TZ)
+      if (data.net_asset_value_date !== expectedDate) {
+        // We should retry fetching DP until we get fresh data or the time is after 10:30 AM ET
+        if (isBeforeTime('10:30:00', TZ)) {
+          logger.warn(
+            `Expected last update - ${expectedDate}, actual ${
+              data.net_asset_value_date
+            }. Retry attempt ${retryCount + 1}. Retrying after ${RETRY_INTERVAL_MINUTES} minute(s)`,
+          )
+          retryCount++
+          setTimeout(() => this.execute(fundId, retryCount), RETRY_INTERVAL_MINUTES * 60 * 1000)
+          // We don't `return` here and let the value be stored in cache on purpose.
+          // This way the EA will respond with the latest value from DP (even though it's not the value that the EA expects),
+          // while it tries to get a fresh update.
+        } else {
+          logger.warn(
+            `Max retires reached. Expected update - ${expectedDate}. Latest update - ${data.net_asset_value_date}.`,
+          )
+        }
+      }
+    }
+
+    const response = {
+      data: {
+        result,
+      },
+      result,
+      statusCode: 200,
+      timestamps: {
+        providerDataRequestedUnixMs,
+        providerDataReceivedUnixMs,
+        providerIndicatedTimeUnixMs: undefined,
+      },
+    }
+    await this.responseCache.write(this.name, [{ params: { fundId }, response }])
+    return response
+  }
+
+  async makeRequest(fundId: number) {
+    const { startDate, endDate } = getStartingAndEndingDates(this.settings.LOOKBACK_DAYS)
+    const requestConfig = {
+      baseURL: this.settings.API_ENDPOINT,
+      url: `funds/${fundId}/nav-daily`,
+      params: {
+        start_date: startDate,
+        end_date: endDate,
+      },
+    }
+
+    const reqKey = calculateHttpRequestKey<BaseEndpointTypes>({
+      context: {
+        adapterSettings: this.settings,
+        inputParameters,
+        endpointName: this.endpointName,
+      },
+      data: requestConfig,
+      transportName: this.name,
+    })
+
+    const { response } = await this.requester.request<ResponseSchema[]>(reqKey, requestConfig)
+    return response
+  }
+
+  // Determines whether we should check for stale data
+  shouldCheckForExpectedData() {
+    // No need to check on weekends, the data from DP will be the same
+    if (isWeekend(new Date())) return false
+
+    // If it's a business day we need to check for stale data only if the current ET time is within certain time range (retry period).
+    // If it's before 09:09 AM we don't need to check as it's too soon and there will be no update from DP,
+    // the scheduler will call the execute function once it's 09:09.
+    // If it's after 10:30 AM, it's too late, and we won't update the value until the next business day.
```

Comment:
Why is after 10:30 too late, will the API not return the data? If so, how does this work for EA's started outside of this ~1.5 hour window?

Reply:
I think this is what DP [suggests](https://smartcontract-it.atlassian.net/browse/DF-19929?focusedCommentId=240523) for their API. It seems like after 10:30 there will be no updates.

Reply:
So any EA that is started after 10:30 ET will fail until the next business day 09:09 ET?

Reply:
EAs started/restarted after 10:30 will behave differently based on couple of factors. 
1. EA is REstarted and already has a cache in redis (no issues)
2 EA is started or is REstarted with no redis cache, then there are two cases 
   2.1 The initial response it gets and caches is the same as with other EAs that saved a response before 10:30 (no issues)
   2.2 The initial response it gets and caches is different from EAs that saved a response before 10:30 (might be issues)
Basically the last point (2.2) means that the both conditions ( DP updated their API and EA (re)started with no redis cache) happened after 10:30. I think DP ensures us that there won't updates after 10:30, that's why they don't want us to retry after that time. In a very unlikely scenario of this, the on-chain price will not be updated as long as the median price reported by most of EAs is the same, so even if we apply case _2.2_ to one or two EAs, this should not trigger on-chain update (this is purely my understanding of on-chain updates based on deviation threshold, can be totally wrong).

Reply:
To clarify, we don't have any 'failure' cases. The EA will save any response it gets from DP, whether it's before 10:30 via scheduler or after that time via initial request (EA restart). The only problem is that under some circumstances the restarted or newly started EAs after 10:30  will have different values in their cache until either all others EAs restart and get that new value or wait for next scheduler execution.